the-new-octal/src/app/service/base.service.ts

```diff
@@ -56,6 +56,29 @@ export class BaseService<T extends {id: number, [key: string]: any}> {
     );
     }
 
+    sum(property: string, value?: string | boolean): Observable<number> {
+      if (value) {
+        return this.getAll().pipe(map(item => item.filter((i) => {
+          if (typeof (i[property]) === 'boolean') { return (i[property] == value); }
+          if ((typeof (i[property]) === 'string') && (value !== '')) { return i[property].toLowerCase().includes((value as string).toLowerCase()); }
+          else { return true }
+        }
+        ).length))
+      }
+      else //number
+      {
+        return this.getAll().pipe(map(list => list.reduce((acc, next) => {
+          if (typeof (next[property]) === 'number') return next[property] + acc;
+          if ((typeof (next[property]) === 'string') && (value !== ''))  return ++acc;
+          if (typeof (next[property])) return next[property] + acc; //boolean és true
+
+        }
+          , 0)))
+      }
+
+    }
+
+
```

Comment:
Mivel nem sok mindent tanítottak így a legjobb javaslatom az, hogy a szummázó metódus egy pipe legyen, aminek az egyik paramétere a lista, a második egy lista azon mezőkről amit szummázni kell. (A lista elhagyható, hogyha ügyesen fejlesztesz)
Visszatérési értéke vagy egy objektum legyen mezőnév: szumma összerendeléssel
Vagy Lista a szumma értékekkel. 

A lista lekérdezést meg list$ | async as-el rakd át egy list-be és arra menjen a pipe.

Reply:
Próbáltam valamit.